contracts/orbital-core/src/orbital_domain.rs

```diff
@@ -0,0 +1,122 @@
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{
+    ensure, Api, Binary, Coin, MessageInfo, QueryRequest, StdError, StdResult, Uint64,
+};
+use neutron_sdk::bindings::{msg::NeutronMsg, query::NeutronQuery};
+
+use crate::{
+    contract::ExecuteDeps, error::ContractError, state::OrbitalDomainConfig,
+    utils::assert_fee_payment,
+};
+
+#[cw_serde]
+pub enum UncheckedOrbitalDomainConfig {
+    Polytone {
+        note: String,
+        timeout: Uint64,
+    },
+    InterchainAccount {
+        connection_id: String,
+        channel_id: String,
+        timeout: Uint64,
+    },
+}
+
+impl UncheckedOrbitalDomainConfig {
+    pub fn try_into_checked(self, api: &dyn Api) -> StdResult<OrbitalDomainConfig> {
```

Comment:
I actually remember thinking between Result and StdResult here and just ended up returning StdResult because that's what the `api.addr_validate` call returns. You think it's worth changing?